draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,545 @@ author:
+  if the TargetName of a SVCB AliasMode record does not resolve to any
+  ServiceMode records, the Proxy-DNS-SVCB list SHALL include a single
+  item containing the AliasMode TargetName and "priority=0".
+
+* If no SVCB records are found authoritatively (i.e., a "NOERROR" or
+  "NXDOMAIN" response when resolving the SVCB DNS query name specified
+  in the Proxy-DNS-Request header), the list MUST contain a single
+  entry with the item value of ".", indicating that the name has no
+  SVCB records.  This special value of "." MUST NOT be returned in
+  error conditions such as timeouts.  The value of the "ttl" parameter
+  MUST be the TTL of this authoritative DNS response.
+
+* If an error condition is encountered (such as a timeout, loop, or
+  invalid record), the Proxy-SVCB-DNS header MUST contain an empty list
+  (and thus not be returned).
+
+
+### Example Proxy-DNS-SVCB response
+
+As an example, assume that the server resolved the SVCB DNS query name
+"example.com" and RR type 65 (HTTPS) as follows:
+
+~~~ diagram
+   example.com.     7200 IN HTTPS 0 foo.svc.example.net.
+   foo.svc.example.net 1800 IN CNAME svc.example.net.
+   svc.example.net. 3600 IN HTTPS 1 svc2.example.net. alpn=h2,h3 echconfig="123..."
+   svc.example.net. 3600 IN HTTPS 2 . alpn=h2 echconfig="abc..."
+~~~
+
+A successful CONNECT response would include the following headers, if the client requested both
+"alpn" and "echconfig":
+
+~~~ example
+HEADERS
+:method = CONNECT
+:status = 200
+proxy-dns-svcb = "svc2.example.net.";priority=1;ttl=1800;key1=:aDIsaDM=:;key5=:MTIzLi4u:,
+                 "svc.example.net.";priority=2;ttl=1800;key1=:aDI=:;key5=:YWJjLi4u:
+~~~
+
+
+## Proxy-DNS-Used Response Header Field {#proxy-dns-used}
+
+A proxy server that receives a request with "Proxy-DNS-Request" with MAY respond with
+the Structured Header "Proxy-DNS-Used" response header fields. The value of
+"Proxy-DNS-Used" MUST be an sf-list whose members are sf-string, each of
+which MUST contain parameters.
+
+~~~ abnf
+Proxy-DNS-Used = sf-list
+~~~
+
+The intent of this header is to provide clients with enough
+information to implement Section 5 of {{SVCB}}.
+
+Proxy servers MUST NOT include the "Proxy-DNS-Used" response header field if the
+corresponding request did not include a "Proxy-DNS-Request" or if its "u"
+parameter had the value sf-false.
+
+Each list member is an sf-string that represents either the IP address
+(IPv4 or IPv6) that the proxy connected to for the CONNECT* request,
+or the value of a DNS CNAME on the path of aliases involved in
+resolving the hostname connected to by the proxy.
+
+The last member in the list MUST have an item value that is the string
+representation of the IPv6 or IPv4 address that represents the
+destination of the CONNECT* from the proxy.  If this is an IPv6
+address, it MUST follow the canonical string form from {{!RFC5952}}.
+
+*TODO*: Is there an IPv4 canonical form RFC we should reference here?
```

Comment:
I don't think so? Closest I could find was https://tools.ietf.org/html/draft-main-ipaddr-text-rep-02

Reply:
Proposed some text above.  I couldn't find any either, and raising this 
to sunset4 doesn't seem like a good use of time.  ;)